File: src/main/java/com/example/demo/configure/MapperScannerConfig.java
package com.example.demo.configure;

import java.util.Properties;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import tk.mybatis.spring.mapper.MapperScannerConfigurer;
/**
 * 
 * @author sxx
 *
 */
@Configuration
public class MapperScannerConfig {
	@Bean
    public MapperScannerConfigurer mapperScannerConfigurer() {
        MapperScannerConfigurer mapperScannerConfigurer = new MapperScannerConfigurer();
        mapperScannerConfigurer.setSqlSessionFactoryBeanName("sqlSessionFactory");
        mapperScannerConfigurer.setBasePackage("com.example.demo.dao");
        Properties properties = new Properties();
        properties.setProperty("mappers", "com.example.demo.utills.BaseDao");
        properties.setProperty("notEmpty", "false");
        properties.setProperty("IDENTITY", "MYSQL");
        mapperScannerConfigurer.setProperties(properties);
        return mapperScannerConfigurer;
    }
}

File: src/main/java/com/example/demo/configure/MapperScannerConfig.kt
package com.example.demo.configure

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import tk.mybatis.spring.mapper.MapperScannerConfigurer
import java.util.*

/**
 *
 * @author luhenchang
 */
@Configuration
class MapperScannerConfig {
    @Bean
    fun mapperScannerConfigurer(): MapperScannerConfigurer {
        val mapperScannerConfigurer = MapperScannerConfigurer()
        mapperScannerConfigurer.setSqlSessionFactoryBeanName("sqlSessionFactory")
        mapperScannerConfigurer.setBasePackage("com.example.demo.dao")
        val properties = Properties()
        properties.setProperty("mappers", "com.example.demo.utills.BaseDao")
        properties.setProperty("notEmpty", "false")
        properties.setProperty("IDENTITY", "MYSQL")
        mapperScannerConfigurer.setProperties(properties)
        return mapperScannerConfigurer
    }
}
